jest.config.js

```diff
@@ -176,9 +176,10 @@ module.exports = {
   // transform: null,
 
   // An array of regexp pattern strings that are matched against all source file paths, matched files will skip transformation
-  // transformIgnorePatterns: [
-  //   "/node_modules/"
-  // ],
+  transformIgnorePatterns: [
+    "/node_modules/",
+    "\\.test\\.js"
```

Comment:
is that how you escape in this context? I would have thought that a `.` is escaped `\.` and a `\` would be escaped with `\\` 🤷‍♂

Reply:
these are regex patterns and the `.` is escaped because we want to match a literal `.`, not any single character. then the `\.` has to be escaped because `\` is an escape character in normal javascript strings.

Reply:
i thought you could write regex literals in javascript, so maybe this could be `/\.test\.js/` instead?

Reply:
Apologies for more stupid question regarding his: If the `\` is also escaped doesn't this mean the regex wouldn't it be treated as literal `\`?

If we want to much any string containing the `.test.js` (my understanding) and because we need to escape the `.` (as in general is a wildchar meaning any single character) wouldn't the regex be `\.test\.js`? [[rubular link]](https://rubular.com/r/vhdhJbnoySTyhF)

I'm not saying this doesn't work, I'm just curious ☺️

Reply:
could be, but I pasted it and my IDE auto-escaped it for the language so :man_shrugging:

Reply:
@xoen I did the same as you and figured out the regex i need is: `\.test\.js`, then I pasted it and my IDE found the the language of the file I was pasting it into was .js and escaped it properly for that to end up with `"\\.test\\.js"`

Reply:
life is too short to think about this stuff

Reply:
Are you kidding me? Regular expressions conversations are my favourite past time 🤣

Reply:
@xoen, no. when the javascript file is parsed by node, it reads `\\.` and generates a string object containing the characters `\.` - the regular expression parser then interprets this as 'literal `.`', rather than 'wildcard single character'.
If the javascript string was `\.`, then javascript would interpret it as `.` and then the regular expression parser would match files named `xtestxjs`.

Reply:
Thanks @andyhd - OMG that's strange. I didn't realise node was also parsing this before, hence the double escape. Thanks very useful.